### PR TITLE
improve geometric models fitting (line, circle) using LSM

### PR DESCRIPTION
**Pull request by Borda**
## Description
improve geometric models fitting (line, circle) using LSM, it closes line #2427 and circle #2428

## Checklist
[It's fine to submit PRs which are a work in progress! But before they are merged, all PRs should provide:]
- [x] Clean style in [the spirit of PEP8](https://www.python.org/dev/peps/pep-0008/)
- [x] [Docstrings for all functions](https://github.com/numpy/numpy/blob/master/doc/example.py)
- [x] Unit tests

## References
- http://www.had2know.com/academics/best-fit-circle-least-squares.html

### CODE REVIEW

**Comment by Borda** [comment]
@sciunto I think, ready for a review...

**Comment by Borda** [comment]
@JDWarner it was used there before, but if you think (r, c) is better I can change it...

File: skimage/measure/fit.py [hunk 1]
```diff
@@ -195,31 +226,29 @@ def estimate(self, data):
         success : bool
             True, if model estimation succeeds.
         """
-
         _check_data_atleast_2D(data)
-
-        X0 = data.mean(axis=0)
+        # https://www.youtube.com/watch?v=puVoOw3hNGY
```
Comment (soupault) [inline skimage/measure/fit.py]: This is something new :laughing:

**Comment by Borda** [comment]
@soupault thanks :)

**Comment by Borda** [comment]
@soupault thanks, so can we merge it? :)

**Comment by sciunto** [comment]
@Borda We are preparing the next release 0.13. This PR will be merged after that. At some point, we need to stop including new stuffs, otherwise we never get to the point where we can release a new version.

I label it correctly, so we will merge it asap. Thanks for your understanding.

**Comment by sciunto** [comment]
@jni How about this one? There is a conflict that must be fixed though...

**Comment by jni** [comment]
@Borda Can you rebase on the current master?

**Comment by sciunto** [comment]
@Borda Travis is broken...

**Comment by Borda** [comment]
it seems that by mistake, I squash the conflict resolving to another commit... :(

**Comment by sciunto** [comment]
Thank you @Borda . I merge.

**Pull request merged**